stream_alert/rule_processor/alert_forward.py

```diff
@@ -0,0 +1,199 @@
+import json
+import os
+import time
+
+import backoff
+import boto3
+from botocore.exceptions import ClientError
+
+from stream_alert.shared import backoff_handlers
+from stream_alert.rule_processor import LOGGER
+
+
+class AlertForwarder(object):
+    """Sends alerts to the Alert Processor and the alerts Dynamo table."""
+    # TODO: Do not send to Alert Processor after Alert Merger is implemented
+    BACKOFF_MAX_RETRIES = 6
+
+    def __init__(self, env):
+        """Initialize the Forwarder with the boto3 clients and resource names.
+
+        Args:
+            env (dict): loaded dictionary containing environment information
+        """
+        self.env = env
+        self.client_dynamo = boto3.client('dynamodb', region_name=self.env['lambda_region'])
+        self.client_lambda = boto3.client('lambda', region_name=self.env['lambda_region'])
+        self.function = os.environ['ALERT_PROCESSOR']
+        self.table = os.environ['ALERT_TABLE']
+
+        # Keep track of unprocessed items when retrying batch_write_item()
+        self.unprocessed_items = None
+
+    def _send_to_lambda(self, alerts):
+        """DEPRECATED: Invoke Alert Processor directly
+
+        Sends a message to the alert processor with the following JSON format:
+            {
+                "record": record,
+                "metadata": {
+                    "rule_name": rule.rule_name,
+                    "rule_description": rule.rule_function.__doc__,
+                    "log": str(payload.log_source),
+                    "outputs": rule.outputs,
+                    "type": payload.type,
+                    "source": {
+                        "service": payload.service,
+                        "entity": payload.entity
+                    }
+                }
+            }
+        """
+        for alert in alerts:
+            try:
+                data = json.dumps(alert, default=lambda o: o.__dict__)
+            except AttributeError as err:
+                LOGGER.error('An error occurred while dumping alert to JSON: %s '
+                             'Alert: %s',
+                             err.message,
+                             alert)
+                continue
+
+            try:
+                response = self.client_lambda.invoke(
+                    FunctionName=self.function,
+                    InvocationType='Event',
+                    Payload=data,
+                    Qualifier='production'
+                )
+
+            except ClientError as err:
+                LOGGER.exception('An error occurred while sending alert to '
+                                 '\'%s:production\'. Error is: %s. Alert: %s',
+                                 self.function,
+                                 err.response,
+                                 data)
+                continue
+
+            if response['ResponseMetadata']['HTTPStatusCode'] != 202:
+                LOGGER.error('Failed to send alert to \'%s\': %s',
```

Comment:
side note: this should really add to a metric instead of only logging

Reply:
Again, this will be deleted in a few days once we've tested the table and added a Lambda function to dispatch alerts from the table